internal/component/otelcol/receiver/solace/solace.go

```diff
@@ -0,0 +1,149 @@
+- Connect Alloy with the following config:
+	otelcol.receiver.solace "solace" {
+		queue = "queue://#telemetry-testprofile"
+		broker = "localhost:5672"
+		auth {
+			sasl_plain {
+				username = "alloy"
+				password = "alloy"
+			}
+		}
+		tls {
+			insecure             = true
+			insecure_skip_verify = true
+		}
+		output {
+			traces = [otelcol.exporter.debug.solace.input]
+		}
+	}
+
+	otelcol.exporter.debug "solace" {
+		verbosity = "detailed"
+	}
+
+	logging {
+		level = "debug"
+	}
+- In "Try Me!", connect as "bob" (username and password)
+- You should be able to see the span in the terminal
+*/
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/grafana/alloy/internal/component"
+	"github.com/grafana/alloy/internal/component/otelcol"
+	otelcolCfg "github.com/grafana/alloy/internal/component/otelcol/config"
+	"github.com/grafana/alloy/internal/component/otelcol/receiver"
+	"github.com/grafana/alloy/internal/featuregate"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/solacereceiver"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	otelextension "go.opentelemetry.io/collector/extension"
+	"go.opentelemetry.io/collector/pipeline"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:      "otelcol.receiver.solace",
+		Stability: featuregate.StabilityGenerallyAvailable,
+		Args:      Arguments{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			fact := solacereceiver.NewFactory()
+			return receiver.New(opts, fact, args.(Arguments))
+		},
+	})
+}
+
+// Arguments configures the otelcol.receiver.solace component.
+type Arguments struct {
+	// The upstream component uses a list for the broker but they only use the first element in the list so I decided to use
+	// a simple string in Alloy to avoid confusing the users.
+	Broker     string `alloy:"broker,attr,optional"`
+	Queue      string `alloy:"queue,attr"`
+	MaxUnacked int32  `alloy:"max_unacknowledged,attr,optional"`
+
+	TLS          otelcol.TLSClientArguments       `alloy:"tls,block,optional"`
+	Flow         FlowControl                      `alloy:"flow_control,block,optional"`
+	DebugMetrics otelcolCfg.DebugMetricsArguments `alloy:"debug_metrics,block,optional"`
+	Auth         Authentication                   `alloy:"auth,block"`
+
+	// Output configures where to send received data. Required.
+	Output *otelcol.ConsumerArguments `alloy:"output,block"`
+}
+
+var _ receiver.Arguments = Arguments{}
+
+// SetToDefault implements syntax.Defaulter.
+func (args *Arguments) SetToDefault() {
+	*args = Arguments{
+		Broker:     "localhost:5671",
+		MaxUnacked: 1000,
+	}
+	args.Flow.SetToDefault()
+	args.DebugMetrics.SetToDefault()
+}
+
+// Validate implements syntax.Validator.
+func (args *Arguments) Validate() error {
+	if args.Auth.PlainText == nil && args.Auth.External == nil && args.Auth.XAuth2 == nil {
```

Comment:
Should we validate that only one form of auth is provided? Also possibly worth upstreaming if we do choose to implement.

Looking at the upstream receiver the toAMPQAuthentication function in messaging_service.go uses PlainText if provided, then XAuth2, then External. If multiple are provided only one is used, though the config Validate upstream also doesn't error with multiple authentication forms provided.

Reply:
Makes sense, I made the change here: https://github.com/grafana/alloy/pull/2084/commits/2c625213441551452cd45462455808453f5dcdee
Also makes sense to contribute upstream, I will create an issue in our board when this get merged to track the contribution

Reply:
Upstream contribution: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36386